db/models/reviews.models.js

```diff
@@ -60,10 +60,19 @@ exports.updateReviewVotes = (id, votesToAdd) => {
 };
 
 exports.fetchCommentsByReviewId = (id) => {
-  return db
+  const validCheck = db.query(
+    `SELECT review_id FROM reviews WHERE review_id = $1;`,
+    [id]
+  );
+  const comments = db
     .query(
       `SELECT * FROM comments WHERE review_id = $1 ORDER BY created_at DESC;`,
       [id]
     )
     .then(({ rows: comments }) => comments);
+  return Promise.all([validCheck, comments]).then(([result, comments]) => {
+    if (result.rows.length < 1)
+      return Promise.reject({ status: 404, msg: "Review does not exist" });
+    return comments;
```

Comment:
What is results wasn't 0....but comments.length was 0? This would be a 200 as it's a valid request. You need to write a test fot this.

Reply:
@two-triangles Wouldnt that be covered by the test on line 165?
results wouldn't be 0 because its a valid id but comments.length would be 0 because there are no comments for it.
Or am i misunderstanding?